test/functional/tests/ci/test_incremental_load.py

```diff
@@ -0,0 +1,146 @@
+    """
+    with TestRun.step("Prepare devices."):
+        cache_disk = TestRun.disks["cache"]
+        cache_disk.create_partitions([Size(2, Unit.GibiByte)])
+        cache_dev = cache_disk.partitions[0]
+        core_disk = TestRun.disks["core"]
+        core_disk.create_partitions([Size(4, Unit.GibiByte)] * 3)
+        core_devs = core_disk.partitions
+        cache_id = 1
+        core_ids = [1, 2, 3]
+
+    with TestRun.step("Start cache."):
+        cache = casadm.start_cache(cache_dev, cache_id=cache_id)
+        if cache.get_status() is not CacheStatus.running:
+            TestRun.fail(f"Cache {cache.core_id} should be running but is {cache.get_status()}.")
+
+    with TestRun.step("Add cores."):
+        for core_dev in core_devs:
+            core = cache.add_core(core_dev)
+            if core.get_status() is not CoreStatus.active:
+                TestRun.fail(f"Core {core.core_id} should be active but is {core.get_status()}.")
+        hidden_core = cache.get_core_devices()[2]
+
+    with TestRun.step("Stop cache."):
+        cache.stop()
+
+    with TestRun.step("Hide core part form from being loaded"):
+        core_disk.remove_partitions()
+        core_disk.create_partitions([Size(4, Unit.GibiByte)] * 2)
+
+    with TestRun.step("Load cache"):
+        cache = casadm.load_cache(cache_dev)
+        if cache.get_status() is not CacheStatus.incomplete:
+            TestRun.fail(f"Cache {cache.cache_id} should be running but is {cache.get_status()}.")
+
+        for core in cache.get_core_devices():
+            if core.get_status() is not CoreStatus.active:
+                TestRun.fail(f"Core {core.core_id} should be Active but is {core.get_status()}.")
+
+        if hidden_core.get_status() is not CoreStatus.inactive:
+            TestRun.fail(f"Hidden core should be Inactive but is {hidden_core.get_status()}.")
+
+    with TestRun.step("Stop cache."):
+        cache.stop(no_data_flush=True)
+
+    with TestRun.step("Recreate core disk partitions."):
+        core_disk.remove_partitions()
+        core_disk.create_partitions([Size(4, Unit.GibiByte)] * 3)
+        core_devs = core_disk.partitions
+
+    with TestRun.step("Add cores to core pool."):
+        for core_dev, core_id in zip(core_devs, core_ids):
+            core = casadm.try_add(core_device=core_dev, cache_id=cache_id, core_id=core_id)
+            if core.get_status() is not CoreStatus.detached:
+                TestRun.fail(f"Core {core.core_id} should be detached but is {core.get_status()}.")
+
+    with TestRun.step("Load cache"):
+        cache = casadm.load_cache(cache_dev)
+        if cache.get_status() is not CacheStatus.running:
+            TestRun.fail(f"Cache {cache.cache_id} should be running but is {cache.get_status()}.")
+
+        cores = cache.get_core_devices()
+        for core in cores:
+            if core.get_status() is not CoreStatus.active:
+                TestRun.fail(f"Core {core.core_id} should be active but is {core.get_status()}.")
```

Comment:
Is this any different than what is done in the previous test?

Reply:
You suggest to pack it into helper function?
It checks if load run successfully after restoring hidden core.

Reply:
But I see we could leave in this test only negative case when core is hidden (first part of that test) because this as you mentioned is covered in first one.